src/main/scala/strawman/collection/View.scala

```diff
@@ -205,6 +205,65 @@ object View extends IterableFactory[View] {
       }
   }
 
+  /*
+    // Implementated as bellow in scala.collection.SeqLike
+    def patch[B >: A, That](from: Int, patch: GenSeq[B], replaced: Int)(implicit bf: CanBuildFrom[Repr, B, That]): That = {
+      val b = bf(repr)
+      var i = 0
+      val it = this.iterator
+      while (i < from && it.hasNext) {
+        b += it.next()
+        i += 1
+      }
+      b ++= patch.seq
+      i = replaced
+      while (i > 0 && it.hasNext) {
+        it.next()
+        i -= 1
+      }
+      while (it.hasNext) b += it.next()
+      b.result()
+    }
+   */
+
+  case class Patched[A](underlying: Iterable[A], from: Int, other: IterableOnce[A], replaced: Int) extends View[A] {
+    if (from < 0 || from > size) throw new IndexOutOfBoundsException(from.toString)
+    def iterator(): Iterator[A] =
+      new Iterator[A] {
+        private[this] var uit = underlying.iterator()
+        private[this] var oit = other.iterator()
+        private[this] var i = 0
+        private[this] var isCurrent = false
+        private[this] var current: A = _
+        def hasNext: Boolean = isCurrent || {
+          if (i < from && uit.hasNext) {
+            i += 1
+            current = uit.next()
+            isCurrent = true
+            true
+          } else {
+            if (i == from) uit.drop(replaced)
+            i += 1
+            if (oit.hasNext) {
+              current = oit.next()
+              isCurrent = true
+              true
+            } else if (uit.hasNext) {
+              current = uit.next()
+              isCurrent = true
+              true
+            } else false
+          }
+        }
```

Comment:
Your implementation doesn’t seem to behave exactly like in the current collections: you may copy more than `replaced` elements if the `other` collection contains more elements.